store/presenters/presenters.go

```diff
@@ -488,7 +489,34 @@ func (a NewAccount) GetName() string {
 
 // Tx is a jsonapi wrapper for an Ethereum Transaction.
 type Tx struct {
-	*models.Tx
+	Confirmed bool           `json:"confirmed"`
+	Data      hexutil.Bytes  `json:"data"`
+	From      common.Address `json:"from"`
+	GasLimit  string         `json:"gasLimit"`
+	GasPrice  string         `json:"gasPrice"`
+	Hash      common.Hash    `json:"hash"`
+	Hex       string         `json:"rawHex"`
+	Nonce     string         `json:"nonce"`
+	SentAt    string         `json:"sentAt"`
+	To        common.Address `json:"to"`
+	Value     string         `json:"value"`
+}
+
+// NewTx builds a transaction presenter.
+func NewTx(tx *models.Tx) Tx {
+	return Tx{
+		Confirmed: tx.Confirmed,
+		Data:      hexutil.Bytes(tx.Data),
```

Comment:
Are these "casts" needed? Both structs should have the same type.

Reply:
You can actually see what the JSON marshaling and unmarshaling look like if we don't make them separate types in [the second commit in this PR](https://github.com/smartcontractkit/chainlink/pull/947/commits/919259ebf712bc6a67c4b3aff49ec5bcb132303c#diff-1ff93c0482d8d63d4388a5145fca5250L525). There's a lot more coercion logic split between the two. By using different types in the presenters, those types have their own JSON marshaling logic and the Tx presenter is much simpler. Hat tip to @dimroc for the suggestion, I personally think it's cleaner this way.

Reply:
Oh, then they do have different types.